CMakeLists.txt

```diff
@@ -0,0 +1,448 @@
+cmake_minimum_required(VERSION 3.15)
```

Comment:
Can this be an older CMake version? See my similar comment [here](https://github.com/FreeRTOS/FreeRTOS-Kernel/pull/438#discussion_r781376454).

Reply:
CMake 3.15 is needed for FILTER generator expression: https://cmake.org/cmake/help/latest/manual/cmake-generator-expressions.7.html#genex:FILTER.
It is used for FREERTOS_HEAP handling in this PR.

Reply:
We can abstract that out into functions. Just like I have done above with your other generator expressions.

Reply:
I will try to do it. In case if it looks well, I decrease cmake_minimum_required appropriate.

Reply:
No need to create a workaround.  I don't see the value in reimplementing CMake functions. I was simply wondering if we could go older. I didn't realize the FILTER expression came from 3.15.